internal/web_server/web/views/pages/dashboard_pages/user_admin_page.templ

```diff
@@ -4,15 +4,154 @@ import (
 	"github.com/Molnes/Nyhetsjeger/internal/web_server/web/views/components/layout_components"
 	"github.com/Molnes/Nyhetsjeger/internal/web_server/web/views/components/dashboard_components/user_admin"
 	"github.com/Molnes/Nyhetsjeger/internal/models/users/usernames"
+	"net/url"
+	"github.com/Molnes/Nyhetsjeger/internal/web_server/web/views/components/icons"
 )
 
+// onSwap is a js script that run either when the page is loaded or when the htmx:afterSwap event is triggered.
+// It will highlight the rows that have been edited or deleted.
+
+script onSwap() {
+
+	const onSwap = () => {
+		console.log("Swapped");
+		for (const [key, value] of deleteMap.entries()) {
+			var row = document.getElementById(key);
+			if (row) {
+				let parent = row.parentElement;
+				parent.classList.add("bg-red-300", "border-b", "border-red-700");
+				parent.getElementsByClassName("del-button")[0].classList.add("hidden");
+				parent.getElementsByClassName("undo-button")[0].classList.remove("hidden");
+			}
+		}
+
+		for (const [key, value] of editMap.entries()) {
+			var row = document.getElementById(key);
+			if (row) {
+				let values = editMap.get(key);
+				let parent = row.parentElement;
+				parent.getElementsByClassName("undo-button")[0].classList.remove("hidden");
+				row.value = values[1];
+			}
+		}
+	}
+
+	onSwap();
+
+	document.body.addEventListener('htmx:afterSwap', function (event) {
+		onSwap();
+	});
+}
+
+// onSave is a js script that runs when the save button is clicked.
+// It will send the changes to the server and clear the maps and the session storage.
+
+script onSave() {
+	fetch("/api/v1/admin/username", {
+		method: "DELETE",
+		headers: {
+			"Content-Type": "application/json",
+		},
+		body: JSON.stringify(Array.from(deleteMap.values()))
+	}).then(response => {
+		if (response.ok) {
+			deleteMap.clear();
+			//Clear deletemap in sessionStorage.
+			sessionStorage.removeItem("deleteMap");
+		}
+	});
+
+
+	let tableNames = ["adjective-table", "noun-table"];
+	const wordMap = tableNames.reduce((map, tableName, index) => {
+		let wordList = Array.from((editMap.entries()).filter(([key, value]) => tableRowMap.get(key) === tableName)).map(([key, value]) => {
+			return value;
+		});
+		map[tableName] = wordList;
+		return map;
+	}, {});
+
+	fetch("/api/v1/admin/username/edit", {
+		method: "POST",
+		headers: {
+			"Content-Type": "application/json",
+		},
+		body: JSON.stringify(wordMap)
+	}).then(response => {
+		if ( response.ok) {
+			editMap.clear();
+			tableRowMap.clear();
+
+			sessionStorage.removeItem("editMap");
+			sessionStorage.removeItem("tableRowMap");
+			location.reload();
+		}
+	});
+	
+}
+
+// reset is a js script that runs when the reset button is clicked. It will clear the maps and the session storage.
+
+script reset() {
+	let decision = confirm("Er du sikker på at du ønsker å nullstille endringer? Dette kan ikke angres.");
```

Comment:
`nullstille endringene` instead of `nullstille endringer`?

Reply:
done